core/src/main/scala/org/apache/spark/sql/execution/CompactExecRowToMutableRow.scala

```diff
@@ -224,3 +216,20 @@ trait CompactExecRowToMutableRow extends ResultWasNull {
     mutableRow
   }
 }
+
+class ResultNullHolder extends ResultWasNull {
+
+  final var wasNull: Boolean = _
+
+  final val defaultCal = new GregorianCalendar()
```

Comment:
DVD stores encoded date/time that needs a Calendar object to convert to Date object. CompactExecRow.getAsDate/getAsTimestamp require the Calendar for the same reason. Likewise, ResultSet.getDate also requires it. This object avoids creating for every column value and keeps reusing it after clearing.